packages/cli-platform-android/src/commands/runAndroid/runOnAllDevices.ts

```diff
@@ -148,8 +148,8 @@ function createInstallError(error: Error & {stderr: string}) {
   }
 
   return new CLIError(
-    `Failed to install the app. ${message}`,
-    message.length > 0 ? undefined : error,
+    `Failed to install the app.${message || ''}`,
+    error.message.length > 0 ? undefined : error,
```

Comment:
I don't understand this line. Isn't `error.message` always going to be some non-empty string? What's the point of this edge case?

Reply:
There are cases when error is empty in which empty error is passed to CLIError which causes the stack to be printed. In that case we have absolutely no idea of what went wrong which is why printing the stack trace only makes sense.

For those edge cases, this would be printed :

```
at makeError (/Users/arushikesarwani/cli/node_modules/execa/index.js:174:9)
    at module.exports.sync (/Users/arushikesarwani/cli/node_modules/execa/index.js:338:15)
    at getGradleTasks (/Users/arushikesarwani/cli/packages/cli-platform-android/build/commands/runAndroid/listAndroidTasks.js:53:32)
    at getTaskNames (/Users/arushikesarwani/cli/packages/cli-platform-android/build/commands/runAndroid/getTaskNames.js:21:73)
    at runOnAllDevices (/Users/arushikesarwani/cli/packages/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:61:55)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Command.handleAction (/Users/arushikesarwani/cli/packages/cli/build/index.js:111:9)
 ```

Reply:
Do we know what exactly throws an empty error?